node/lib/util/push.js

```diff
@@ -93,24 +94,23 @@ exports.push = co.wrap(function *(repo, remoteName, source, target, force) {
     // First, push the submodules.
 
     let errorMessage = "";
-    const subRepos = yield SubmoduleUtil.getSubmoduleRepos(repo);
     const shas = yield SubmoduleUtil.getSubmoduleShasForBranch(repo, source);
-    const index = yield repo.index();
-    const urls = yield SubmoduleConfigUtil.getSubmodulesFromIndex(repo, index);
```

Comment:
Apparently, according to the git mailing list discussion I read earlier today, you should be using the one from the config (or the configured remote).  That's to allow old versions with bad URLs to still work.